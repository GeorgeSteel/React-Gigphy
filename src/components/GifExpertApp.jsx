import React, { useState } from 'react'
import AddCategory from './AddCategory'
import GifGrid from './GifGrid'

export default function GifExpertApp() {
  const [categories, setCategories] = useState(['One Punch Man'])

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map(categorie => (
          <GifGrid key={categorie} category={categorie} />
        ))}
      </ol>
    </>
  )
}
