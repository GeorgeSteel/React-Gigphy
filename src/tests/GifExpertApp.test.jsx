import React from 'react'

import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import GifExpertApp from '../components/GifExpertApp'

describe('Root Component Tests', () => {
  let wrapper = shallow(<GifExpertApp />)

  beforeEach(() => (wrapper = shallow(<GifExpertApp />)))

  test('should show the main component without issues', () => {
    wrapper = shallow(<GifExpertApp />)

    expect(wrapper).toMatchSnapshot()
  })

  test('should return a categories list', () => {
    const categories = ['Saitama', 'Goku']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
