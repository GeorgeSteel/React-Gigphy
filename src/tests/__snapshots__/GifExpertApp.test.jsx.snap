// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component Tests should return a categories list 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Saitama"
      key="Saitama"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`Root Component Tests should show the main component without issues 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
